### PR TITLE
Prepare angular-imask 5.2.0

### PR DESCRIPTION
**Pull request by listepo-alterpost**
- Code style and type improvements for angular-imask
- The added new build system for angular-imask

### CODE REVIEW

**Comment by uNmAnNeR** [comment]
hi! what is the purpose of PR? something wrong with current solution?

**Comment by listepo-alterpost** [comment]
Should fix https://github.com/uNmAnNeR/imaskjs/issues/115 and some ts errors for angular with ts `strict: true`

**Comment by listepo-alterpost** [comment]
in the lib/imask.directive.ts only code style and type improvements, logic not changed

**Comment by listepo-alterpost** [comment]
only `dist` folder should be published

**Comment by listepo-alterpost** [comment]
`ng-packagr` best for Angular Package Format

**Comment by listepo-alterpost** [comment]
@uNmAnNeR any comments?

**Comment by DanielGBullido** [comment]
the current version (5.1.5) not found in angular. please merge this ASAP!

File: packages/angular-imask/tsconfig.json [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
   "angularCompilerOptions": {
+    "annotateForClosureCompiler": true,
     "skipTemplateCodegen": true,
-    "skipMetadataEmit": false,
-    "strictMetadataEmit" : true,
-    "annotationsAs": "decorators"
+    "strictMetadataEmit": true,
+    "fullTemplateTypeCheck": true,
+    "strictInjectionParameters": true,
+    "enableResourceInlining": true
```
Comment (uNmAnNeR) [inline packages/angular-imask/tsconfig.json]: does not make sense, we don't use styles nor templates

File: packages/angular-imask/tsconfig.json [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
-    "src/index.ts"
-  ],
   "angularCompilerOptions": {
+    "annotateForClosureCompiler": true,
     "skipTemplateCodegen": true,
-    "skipMetadataEmit": false,
-    "strictMetadataEmit" : true,
-    "annotationsAs": "decorators"
+    "strictMetadataEmit": true,
+    "fullTemplateTypeCheck": true,
```
Comment (uNmAnNeR) [inline packages/angular-imask/tsconfig.json]: there is no templates in project, seems to be unnecessary

**Comment by uNmAnNeR** [comment]
Hello guys, thanks for PR and sorry for long delay.
1) https://lerna.js.org/ is used, so to keep automatic version bumping `imask` dep needs to be link to file as it was: `file:...`, and not in peers. Also you add a lot of other dependencies which actually are devDeps, but not runtime.
2) Specific point about using `ng-packagr`. I am not sure if it is really necessary to use. What is the reason to use it? I am trying to avoid unnecessary deps as much as possible which probably not what angular community does in general. Thats the difference and i tend to follow my rule here. So probably i'd like to stay with raw low-level `ngc` and `rollup` which are solid stable.
3) You drop angular<6 support, but still no 8.0.0 support. I can not agree with it. Current condition `>=4.0.0` much more preferable.
4) I don't see any reasons for moving sources from `src` to `lib`.  We've just lost change history with no real benefit.

In general i think that if the only reason was to fix #115 then it's too overengeneered. I prefer to use simple and clear tools which is harder to configurate at some point, but easy to maintain in long perspective. Also i suppose there should be simple way to fix `strict` mode if it really matters.

@DanielGBullido what do you mean saying "not found in angular"?
I use `angular-imask` every day from npm with latest angular with no issues.

**Comment by DanielGBullido** [comment]
@uNmAnNeR  i install de plugin angular-imask in angular 7 and  use :  
<input
        [imask]="{mask: '+{7}(000)000-00-00'}"
      />

but  when  modified input dont change format  :(

**Comment by DanielGBullido** [comment]
@uNmAnNeR sorry, my fault!

**Comment by listepo-alterpost** [comment]
I will try to fix it without major changes.

**Pull request closed without merging**